packages/smart-contracts/scripts-create2/compute-from-list.ts

```diff
@@ -0,0 +1,38 @@
+import { computeCreate2DeploymentAddress } from './compute-one-address';
+import { create2ContractDeploymentList } from './utils';
+import { HardhatRuntimeEnvironmentExtended } from './types';
+
+export const computeCreate2DeploymentAddressesFromList = async (
+  hre: HardhatRuntimeEnvironmentExtended,
+): Promise<void> => {
+  if (!hre.config.xdeploy.networks || !hre.config.xdeploy.rpcUrls) {
+    throw new Error('Bad network configuration');
+  }
+
+  if (!hre.config.xdeploy.salt) {
+    throw new Error('Missing salt');
+  }
+
+  if (!hre.config.xdeploy.deployerAddress) {
+    console.warn('Deployer address is set to default !');
+  }
+
```

Comment:
These 3 checks are already done at `xdeploy` level so no need to add them here, I think.